### PR TITLE
Add sendMessageAll function to API

### PR DESCRIPTION
**Pull request by kennedn**
This pull request addresses issue #1, please see issue for details.

### CODE REVIEW

File: include/pico_ws_server/web_socket_server.h [hunk 1]
```diff
@@ -30,6 +30,11 @@ class WebSocketServer {
   // Send a BINARY message
   bool sendMessage(uint32_t conn_id, const void* payload, size_t payload_size);
 
+  // Send a TEXT message to all connections, payload must be a null-terminated string
+  bool sendMessageAll(const char* payload);
```
Comment (cadouthat) [inline include/pico_ws_server/web_socket_server.h:34]: nit: maybe it should be called `broadcastMessage`? This is fine if not

File: src/web_socket_server_internal.cpp [hunk 2]
```diff
@@ -192,6 +192,38 @@ bool WebSocketServerInternal::sendMessage(uint32_t conn_id, const void* payload,
...
+bool WebSocketServerInternal::sendMessageAll(const void* payload, size_t payload_size) {
+  if (connection_by_id.size() == 0) {
+    DEBUG("connection map is empty");
+    return false;
+  }
+
+  cyw43_thread_enter();
+  
+  for (auto& kv: connection_by_id) {
+    kv.second->sendWebSocketMessage(payload, payload_size);
```
Comment (cadouthat) [inline src/web_socket_server_internal.cpp:220]:
```suggestion
  for (const auto& [_, connection] : connection_by_id) {
    connection->sendWebSocketMessage(payload, payload_size);
```

File: src/web_socket_server_internal.cpp [hunk 3]
```diff
@@ -192,6 +192,38 @@ bool WebSocketServerInternal::sendMessage(uint32_t conn_id, const void* payload,
...
+bool WebSocketServerInternal::sendMessageAll(const char* payload) {
+  if (connection_by_id.size() == 0) {
+    DEBUG("connection map is empty");
+    return false;
+  }
+
+  cyw43_thread_enter();
+  
+  for (auto& kv: connection_by_id) {
+    kv.second->sendWebSocketMessage(payload);
```
Comment (cadouthat) [inline src/web_socket_server_internal.cpp:204]:
```suggestion
  for (const auto& [_, connection] : connection_by_id) {
    connection->sendWebSocketMessage(payload);
```

**Review by cadouthat — COMMENTED** [review]
Thanks for the PR!

File: include/pico_ws_server/web_socket_server.h (hunk 1 above)
Comment (kennedn) [reply include/pico_ws_server/web_socket_server.h]: I have renamed sendMessageAll to broadcastMessage now.

File: src/web_socket_server_internal.cpp (hunk 2 above)
Comment (kennedn) [reply src/web_socket_server_internal.cpp]: I have used structured bindings for the loops as suggested.

File: src/web_socket_server_internal.cpp (hunk 3 above)
Comment (kennedn) [reply src/web_socket_server_internal.cpp]: I have used structured bindings for the loops as suggested.

**Comment by kennedn** [comment]
Hey, I have pushed another commit with all your suggested changes. 

Cheers,

**Review by cadouthat — APPROVED** [review]

**Pull request merged**